tests/emukit/sensitivity/test_emukit_sensitivity.py

```diff
@@ -1,12 +1,44 @@
+import mock
 import pytest
 import numpy as np
-import random
 
 from emukit.core import ContinuousParameter, ParameterSpace
-from emukit.sensitivity.monte_carlo import MonteCarloSensitivity
+from emukit.sensitivity.monte_carlo import MonteCarloSensitivity, ModelBasedMonteCarloSensitivity
 from emukit.sensitivity.test_functions import Ishigami
+from emukit.core.interfaces import IModel
 
 
+class MockModel(IModel):
+    pass
+
+@pytest.fixture
+def model():
+    model = mock.create_autospec(MockModel)
+    model.predict.return_value = (0.1*np.ones((3, 5)), 0.5*np.ones((3, 5)))
+
+    return model
+
+def test_model_based_montecarlo_sensitivity(model):
+
+    space = ParameterSpace([ContinuousParameter('x1', -np.pi, np.pi),
+                            ContinuousParameter('x2', -np.pi, np.pi),
+                            ContinuousParameter('x3', -np.pi, np.pi)])
+
+    num_mc = 1000
+    np.random.seed(0)
+    senstivity_ishigami = ModelBasedMonteCarloSensitivity(model, space)
+    senstivity_ishigami.generate_samples(1)
+
+    main_sample = np.array([[ 3.10732573, -1.25504469, -2.66820221], [-1.32105416, -1.45224686, -2.06642419],[ 2.41144646, -1.98949844,  1.66646315]])
+    fixing_sample = np.array([[-2.93034587, -2.62148462,  1.71694805], [-2.70120457,  2.60061313,  3.00826754], [-2.56283615,  2.53817347, -1.00496868]])
+
+    main_effects, total_effects, total_variance = senstivity_ishigami.compute_effects(main_sample = main_sample,fixing_sample=fixing_sample,num_monte_carlo_points = num_mc)
+
+    keys = space.parameter_names
+
+    assert(all(k in main_effects for k in keys))
```

Comment:
As the model gives constant predictions, can we test that the sensitivity indices are what are expected?

Reply:
yap, done.

Reply:
What I meant was that I thought the main effects and total effects should be zero because the model outputs constant predictions. If that is true, we should test it here. If it's not true, don't worry.  In response to this comment did you add the test below? Please discuss with @apaleyes whether we want this kind of test.